app/src/lib/git/branch.ts

```diff
@@ -117,3 +118,21 @@ async function checkIfBranchExistsOnRemote(
   )
   return result.stdout.length > 0
 }
+
+export async function getBranchesPointedAt(
+  repository: Repository,
+  commitish: string
+): Promise<Array<string>> {
+  const args = [
+    'branch',
+    `--points-at=${commitish}`,
+    '--format=%(refname:short)',
+  ]
+  const { stdout } = await git(args, repository.path, 'branchPointedAt')
+  // split along newlines, which should just be branch names
+  const captures = getCaptures(stdout, /\s*(.+)\n/g)
```

Comment:
If we use `--format` to control the output, we can avoid the overhead of regexes here. An advanced example of this is how `for-each-ref.ts` enumerates the list of branches - this feels similar to what you're looking for here: https://github.com/desktop/desktop/blob/f2c460236c9bf2cbfeff3c0df64273316dedc1ce/app/src/lib/git/for-each-ref.ts#L17

Reply:
good call! b2fdd57